src/common/dataAdaptors/dictionaryAdaptors.js

```diff
@@ -0,0 +1,58 @@
+import { relatedMediaForEditing } from 'common/dataAdaptors/relatedMediaAdaptor'
+import { objectsToIdsAdaptor } from 'common/dataAdaptors/objectsToIdsAdaptor'
+import {
+  audienceForEditing,
+  audienceForApi,
+} from 'common/dataAdaptors/audienceAdaptor'
+
+export function entryForEditing(rawEntry) {
+  return {
+    ...entryForViewing(rawEntry),
+    ...relatedMediaForEditing(rawEntry),
+    ...audienceForEditing(rawEntry),
+    partOfSpeech: rawEntry?.partOfSpeech?.id,
+  }
+}
+
+export function entryForViewing(rawEntry) {
+  return {
+    id: rawEntry?.id || '',
+    acknowledgements: rawEntry?.acknowledgements || [],
+    alternateSpellings: rawEntry?.alternateSpellings || [],
+    categories: rawEntry?.categories || [],
+    excludeFromGames: rawEntry?.excludeFromGames,
+    excludeFromKids: rawEntry?.excludeFromKids,
+    notes: rawEntry?.notes || [],
+    partOfSpeech: rawEntry?.partOfSpeech || '',
+    pronunciations: rawEntry?.pronunciations || [],
+    relatedEntries: rawEntry?.relatedDictionaryEntries || [],
+    relatedAudio: rawEntry?.relatedAudio || [],
+    relatedImages: rawEntry?.relatedImages || [],
+    relatedVideos: rawEntry?.relatedVideos || [],
```

Comment:
If you wanted to, I think you could use the relatedMediaAdaptor and notesAcknowledgementsAdaptor for the entryForViewing and entryForApi properties.